site-src/geps/gep-957.md

```diff
@@ -0,0 +1,140 @@
+# GEP-957: Destination Port Matching
+
+* Issue: [#957](https://github.com/kubernetes-sigs/gateway-api/issues/957)
+* Status: Provisional
+
+## TLDR
+
+Add a new `port` field to ParentRef to support port matching in Routes.
+
+## Goals
+
+* Support port matching in routes based on the destination port number of the
+  request.
+
+## Non-Goals
+
+* Support port matching based on named port.
+
+## Introduction
+
+Port matching is a common service mesh use case where traffic policies/rules
+need to be applied to traffic to certain destination ports. For ingress, while
+the API today already supports attaching a route to a specific listener, it may
+be useful to support attaching routes to listener(s) on a specified port. For
+example, some users may only want to expose their application on ports 80 or 443
+on the Gateway.
```

Comment:
> be useful to support attaching routes to listener(s) on a specified port

Why might that be useful?
Please elaborate on that for the purpose of this GEP.

80/443 is not the best example here probably. Readers are not imaginative so picking a better example will bring more clarity.

Reply:
Updated this section and also the field comment in API section to be more prescriptive. 

Admittedly while there are clear, concrete mesh use cases, it's only potentially useful for ingress(@robscott please chime in if I am missing a concrete ingress use case here). I am hoping that this GEP is a reasonable middle ground which enhances this API to be more inclusive towards use cases outside of ingress(such as mesh) while not making it more confusing/complicated for the existing ingress support.